operator/controllers/http/httpscaledobject_controller.go

```diff
@@ -146,7 +148,20 @@ func (r *HTTPScaledObjectReconciler) Reconcile(ctx context.Context, req ctrl.Req
 // SetupWithManager sets up the controller with the Manager.
 func (r *HTTPScaledObjectReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&httpv1alpha1.HTTPScaledObject{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		For(&httpv1alpha1.HTTPScaledObject{}, builder.WithPredicates(
+			predicate.Or(
+				predicate.GenerationChangedPredicate{},
+				util.HTTPScaledObjectReadyConditionPredicate{},
+			),
+		)).
+		// Trigger a reconcile only when the ScaledObject spec,label or annotation changes.
+		// Ignore updates to ScaledObject status
+		Owns(&kedav1alpha1.ScaledObject{}, builder.WithPredicates(
+			predicate.Or(
+				predicate.LabelChangedPredicate{},
+				predicate.AnnotationChangedPredicate{},
```

Comment:
I'm not 100% sure if we should be aware of changes on labels, but I think that we have to be for annotations as we propagate them to the ScaledObject for pausing actions

Reply:
Agree